gammapy/utils/regions.py

```diff
@@ -275,3 +277,94 @@ def region_circle_to_ellipse(region):
         center=region.center, width=region.radius, height=region.radius
     )
     return region_new
+
+
+def containment_region(integral_map, fraction=0.68, n_levels=100, apply_union=True):
+    """Find the iso-contours region corresponding to a given containment
+        for a map of integral quantities.
+
+    Parameters
+    ----------
+    integral_map : `~gammapy.maps.WcsNDMap`
+        Map of integral quantities
+    fraction : float
+        Containment fraction
+    n_levels : int
+        Numbers of contours levels used to find the required containment region.
+
+    Returns
+    -------
+    regions : list of ~regions.PolygonSkyRegion` or `~regions.CompoundSkyRegion`
+        regions from iso-contours matching containment fraction
+    """
+    integral_map = integral_map.reduce_over_axes()
+    fmax = np.nanmax(integral_map.data)
+    if fmax != 0.0:
+        frange = np.linspace(fmax / n_levels, fmax, n_levels)
+        fsum = integral_map.data.sum()
+        for fval in frange:
+            S = np.sum(integral_map.data[integral_map.data > fval]) / fsum
```

Comment:
Same comment as below, you could argsort the `integral_map.data`  and compute the cumulative sum.

Reply:
done it's faster but slightly less precise especially for the radius computation